### ISSUE TITLE
Failed test 'network error'

### ISSUE THREAD

**Issue by eserte** [powerman/perl-MojoX-JSONRPC2-HTTP#2]
t/basic.t started to fail:
```
#   Failed test 'network error'
#   at t/basic.t line 47.
#                   '301 Moved Permanently'
#     doesn't match '(?^umsi:connect|resolve)'
# Looks like you failed 1 test of 22.
t/basic.t ................. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/22 subtests 
```

**Comment by eserte** [comment]
Statistical analysis suggests that this happens with newer Mojolicious versions. There are no pass reports on my smoker systems for Mojolicious 9.19 and newer:
```
****************************************************************
Regression 'mod:Mojolicious'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	4455859719636187.00
[1='eq_8.43']  	     -0.0000	      0.0000	  -2.27
[2='eq_8.73']  	     -0.0000	      0.0000	  -3.24
[3='eq_9.0']   	     -0.0000	      0.0000	  -2.46
[4='eq_9.01']  	     -0.0000	      0.0000	  -2.95
[5='eq_9.19']  	     -1.0000	      0.0000	-3638194226176522.00
[6='eq_9.21']  	     -1.0000	      0.0000	-3638194226176520.50
[7='eq_9.22']  	     -1.0000	      0.0000	-4393536933944594.00
[8='eq_9.23']  	     -1.0000	      0.0000	-4322818888704727.50

R^2= 1.000, N= 153, K= 9
****************************************************************
```

**Comment by powerman** [comment]
Well, if Mojolicious developers broke things they are welcome to open PR with a fix.

Thanks for the report.

**Comment by eserte** [comment]
No, it's not Mojolicious. It looks like an issue with name resolution, the same as reported in https://github.com/tokuhirom/Furl/issues/128 and https://github.com/uperl/Test2-Tools-HTTP/issues/4 . To reproduce the issue everywhere (well, at least Linux systems), try something like
```
LOCALDOMAIN=bing.com ./Build test
```
To fix the issue add a dot to the invalid host name:
```
diff --git a/t/basic.t b/t/basic.t
index 5f58b03..e47a37a 100644
--- a/t/basic.t
+++ b/t/basic.t
@@ -43,7 +43,7 @@ is ref $client->ua, 'Mojo::UserAgent',
     'default ->ua';
 
 # network error
-my ($failed,$result,$error) = $client->url('http://no-such-host-qKfdsEsZ/')->call('method');
+my ($failed,$result,$error) = $client->url('http://no-such-host-qKfdsEsZ./')->call('method');
 like $failed, qr/connect|resolve/msi, 'network error';
 
 $client->url('/');
```

**Comment by eserte** [comment]
And in the meanwhile the statistical analysis looks quite different, R² being far below 1 now:
```
****************************************************************
Regression 'mod:Mojolicious'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.3424	   2.92
[1='eq_8.43']  	      0.0000	      0.4842	   0.00
[2='eq_8.73']  	      0.0000	      0.3529	   0.00
[3='eq_9.0']   	      0.0000	      0.3609	   0.00
[4='eq_9.01']  	      0.0000	      0.3448	   0.00
[5='eq_9.19']  	     -0.8750	      0.3632	  -2.41
[6='eq_9.21']  	     -0.5000	      0.3828	  -1.31
[7='eq_9.22']  	     -0.7170	      0.3456	  -2.07
[8='eq_9.23']  	     -0.5814	      0.3464	  -1.68

R^2= 0.507, N= 206, K= 9
****************************************************************
```
So in this case it's not Mojolicious' fault (but in many other cases it is...).

**Comment by powerman** [comment]
> To fix the issue add a dot to the invalid host name:

Thanks! Fixed in 2.0.3.

> So in this case it's not Mojolicious' fault (but in many other cases it is...).

Sorry if I overreacted a bit - I'm in Kharkov, it's war here, and I really not in the mood for fixing things after Mojo break something once again.
